src/Activity.js

```diff
@@ -1,44 +1,77 @@
-    return userActivityByDate.minutesActive;
-  }
-  calculateActiveAverageForWeek(id, date, userRepo) {
-    return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => {
-      return acc += elem.minutesActive;
-    }, 0) / 7).toFixed(1));
-  }
-  accomplishStepGoal(id, date, userRepo) {
-    let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
-    if (userStepsByDate.numSteps === userRepo.dailyStepGoal) {
+    this.activityData = activityData;
+  };
+
+  calculateMilesWalked(user, date, userRepo) {
+    let foundUser = this.activityData.find(data => user.id === data.userID && date === data.date);
+    let userSteps = foundUser.numSteps;
+    let stepsNeededforOneMile = 5280 / user.strideLength;
+    let milesTraveled = (userSteps / stepsNeededforOneMile).toFixed(2);
+    return Number(milesTraveled);
+  };
+
+  findMinutesActiveAverage(user, date, userRepo) {
+    let foundWeek = userRepo.findWeekOfData(date, user.id, this.activityData);
+    return (foundWeek.reduce((totalMinutes, day) => {
+      return totalMinutes + day.minutesActive
+    }, 0)) / 7;
+  };
+
+  accomplishStepGoal(user, date, userRepo) {
+    let foundActivityByDate = this.activityData.find(activity => activity.userID === user.id && activity.date === date);
+    if (foundActivityByDate.numSteps >= user.dailyStepGoal) {
       return true;
+    } else {
+      return false;
     }
-    return false
-  }
-  getDaysGoalExceeded(id, userRepo) {
-    return this.activityData.filter(data => id === data.userID && data.numSteps > userRepo.dailyStepGoal).map(data => data.date);
-  }
-  getStairRecord(id) {
-    return this.activityData.filter(data => id === data.userID).reduce((acc, elem) => (elem.flightsOfStairs > acc) ? elem.flightsOfStairs : acc, 0);
-  }
-  getAllUserAverageForDay(date, userRepo, relevantData) {
-    let selectedDayData = userRepo.chooseDayDataForAllUsers(this.activityData, date);
-    return parseFloat((selectedDayData.reduce((acc, elem) => acc += elem[relevantData], 0) / selectedDayData.length).toFixed(1));
-  }
-  userDataForToday(id, date, userRepo, relevantData) {
-    let userData = userRepo.getDataFromUserID(id, this.activityData);
-    return userData.find(data => data.date === date)[relevantData];
-  }
-  userDataForWeek(id, date, userRepo, releventData) {
-    return userRepo.getWeekFromDate(date, id, this.activityData).map((data) => `${data.date}: ${data[releventData]}`);
-  }
+  };
+
+  getUserStat(date, user, stat) {
+    let activityByDate = this.activityData.find(activity => activity.date === date && activity.userID === user.id);
+    return activityByDate[stat];
+  };
+
+  getUserTotalsForWeek(user, date, userRepo, stat) {
+    let userWeek = userRepo.findWeekOfData(date, user.id, this.activityData);
+    let statTotal = userWeek.reduce((accu, current) => {
+      accu = accu + current[stat];
+      return accu;
+    }, 0);
+    return statTotal;
+  };
```

Comment:
I like using "accum" and "current" with reduce, but may be best to make them more descriptive.